src/fpm/cmd/search.f90

```diff
@@ -0,0 +1,178 @@
+    private
+    public :: cmd_search
+  
+  contains
+  
+    !> Search the fpm registry for a package
+    subroutine cmd_search(settings)
+        !> Settings for the search command.
+        class(fpm_search_settings), intent(in) :: settings
+        type(fpm_global_settings) :: global_settings
+        character(:), allocatable :: tmp_file, name, namespace, description, query_url, package_version
+        integer :: stat, unit, ii
+        type(json_object) :: json
+        type(json_object), pointer :: p
+        !> Error handling.
+        type(error_t), allocatable :: error
+        type(toml_array), pointer :: array
+        type(version_t), allocatable :: version
+
+        !> Downloader instance.
+        class(downloader_t), allocatable :: downloader
+        allocate (downloader)
+
+        call get_global_settings(global_settings, error)
+        if (allocated(error)) then
+            call fpm_stop(1, "Error retrieving global settings"); return
+        end if
+
+        !> Generate a temporary file to store the downloaded package search data
+        tmp_file = get_temp_filename()
+        open (newunit=unit, file=tmp_file, action='readwrite', iostat=stat)
+        if (stat /= 0) then
+        call fatal_error(error, "Error creating temporary file for downloading package."); return
+        end if
+        query_url = settings%registry//'/packages_cli' &
+                    & // '?query='//settings%query &
+                    & // '&page='//settings%page &
+                    & // '&package='//settings%package &
+                    & // '&namespace='//settings%namespace &
+                    & // '&version='//settings%version &
+                    & // '&license='//settings%license &
+                    & // '&limit='//settings%limit &
+                    & // '&sort_by='//settings%sort_by &
+                    & // '&sort='//settings%sort
+
+        !> Get the package data from the registry
+        call downloader%get_pkg_data(query_url, version, tmp_file, json, error)
+        close (unit)
+        if (allocated(error)) then
+            call fpm_stop(1, "Error retrieving package data from registry: "//settings%registry); return
+        end if
+        print *
+        print *, "Searching packages in Local Registry:"
+        print *
+        call search_package(settings%query, settings%namespace, settings%package, settings%version)
+        if (json%has_key("packages")) then
+            call get_value(json, 'packages', array)
+            print *
+            print '(A,I0,A)', ' Found ', len(array), ' packages in fpm - registry:'
+            print *
+            do ii=1, len(array)
+                call get_value(array, ii, p)
+                call get_value(p, 'name', name)
+                call get_value(p, 'namespace', namespace)
+                call get_value(p, 'description', description)
+                call get_value(p, 'version', package_version)
+
+                print *, "Name: ", name
```

Comment:
What would be a suitable CLI output format for representing the packages ?

Reply:
You can take a look at function `info_package` that returns information about a whole package: 
https://github.com/fortran-lang/fpm/blob/83b4412fdb92e3238fd2f8ac5757b12386b8b51f/src/fpm_model.f90#L247

However, that is meant for debugging purposes. For screen output, some more polished version is needed imho